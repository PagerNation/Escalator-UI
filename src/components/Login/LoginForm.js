import React, { PropTypes } from 'react';
import Logo from  '../../static/escalator.png';
import { Header, Form, Input, Button, Segment, Image, Message } from 'semantic-ui-react';

class LoginForm extends React.Component {

  render() {
    return (
      <div>
        <Image src={Logo} size='tiny' />
        <Header as='h2'>
          Escalator Login
        </Header>
          <Form error={this.props.loginError} onSubmit={this.props.onSubmit}>
          <Segment stacked>
            <Form.Field>
            <Input
            name="email"
            placeholder='Email'
            icon='user'
            iconPosition='left'
            type="email"
            required />
            </Form.Field>
            <Form.Field>
            <Input
            name="password"
            placeholder='Password'
            icon='lock'
            iconPosition='left'
            type="password"
            required />
            </Form.Field>
            <Button primary fluid type='submit'>Submit</Button>
        </Segment>
        <Message
          error
          content='Your login information is incorrect.' />
        </Form>
      </div>
    );
  }
}

LoginForm.PropTypes = {
  loginError: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default LoginForm;
