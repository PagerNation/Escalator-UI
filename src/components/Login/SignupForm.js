import React, { PropTypes } from 'react';
import Logo from  '../../static/escalator.png';
import { Header, Form, Input, Button, Segment, Image, Message } from 'semantic-ui-react';

class SignupForm extends React.Component {

  render() {
    return (
      <div>
        <Image src={Logo} size='tiny' />
        <Header as='h2'>
          Create an Account
        </Header>
        <Form error={!!this.props.error} onSubmit={this.props.onSubmit}>
          <Segment stacked>
              <Form.Field>
                <Input
                  name="name"
                  placeholder='Name'
                  required />
              </Form.Field>
              <Form.Field>
                <Input
                  name="email"
                  placeholder='Email'
                  type="email"
                  required />
              </Form.Field>
              <Form.Field>
                <Input
                  name="password"
                  placeholder='Password'
                  type="password"
                  required />
              </Form.Field>
              <Form.Field>
                <Input
                  name="password2"
                  placeholder='Confirm Password'
                  type="password"
                  required />
              </Form.Field>
              <Button primary fluid type='submit'>Submit</Button>
          </Segment>
          <Message
            error
            content={this.props.error}
          />
        </Form>
      </div>
    );
  }
}

SignupForm.PropTypes = {
  error: PropTypes.bool.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default SignupForm;
