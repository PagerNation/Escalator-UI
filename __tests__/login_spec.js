import React from 'react';
import LoginForm from '../src/components/Login/LoginForm';
import SignupForm from '../src/components/Login/SignupForm';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Login', () => {
  it('renders LoginForm', () => {
    const component = renderer.create(<LoginForm />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })

  it('renders SignupForm', () => {
    const component = renderer.create(<SignupForm />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
