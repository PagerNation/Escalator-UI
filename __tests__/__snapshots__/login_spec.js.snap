// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders LoginForm 1`] = `
<div>
  <img
    alt={undefined}
    className="ui tiny image"
    height={undefined}
    src="test-file-stub"
    width={undefined}
  />
  <h2
    className="ui header"
  >
    Escalator Login
  </h2>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="ui stacked segment"
    >
      <div
        className="field"
      >
        <div
          className="ui left icon input"
        >
          <i
            className="user icon"
          />
          <input
            name="email"
            placeholder="Email"
            required={true}
            type="email"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui left icon input"
        >
          <i
            className="lock icon"
          />
          <input
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
      </div>
      <button
        className="ui fluid primary button"
        tabIndex={undefined}
        type="submit"
      >
        Submit
      </button>
    </div>
    <div
      className="ui error message"
    >
      <div
        className="content"
      >
        <p>
          Your login information is incorrect.
        </p>
      </div>
    </div>
  </form>
</div>
`;

exports[`Login renders SignupForm 1`] = `
<div>
  <img
    alt={undefined}
    className="ui tiny image"
    height={undefined}
    src="test-file-stub"
    width={undefined}
  />
  <h2
    className="ui header"
  >
    Create an Account
  </h2>
  <form
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="ui stacked segment"
    >
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            name="name"
            placeholder="Name"
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            name="email"
            placeholder="Email"
            required={true}
            type="email"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            name="password"
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            name="password2"
            placeholder="Confirm Password"
            required={true}
            type="password"
          />
        </div>
      </div>
      <button
        className="ui fluid primary button"
        tabIndex={undefined}
        type="submit"
      >
        Submit
      </button>
    </div>
    <div
      className="ui error message"
    />
  </form>
</div>
`;
